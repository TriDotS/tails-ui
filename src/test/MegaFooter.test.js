import MegaFooter from '../components/MegaFooter'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <MegaFooter
      copyrightText="Copyright 2017 Tyler Knipfer"
      columnOne={{
        header: 'Links',
        links: [
          { key: 'faq', value: <a href="#!">FAQ</a> },
          { key: 'about', value: <a href="#!">About</a> },
          { key: 'support', value: <a href="#!">Support</a> }
        ]
      }}
    />,
    div
  )
})

test('mega footer snapshot', () => {
  const tree = renderer
    .create(
      <MegaFooter
        copyrightText="Copyright 2017 Tyler Knipfer"
        columnOne={{
          header: 'Links',
          links: [
            { key: 'faq', value: <a href="#!">FAQ</a> },
            { key: 'about', value: <a href="#!">About</a> },
            { key: 'support', value: <a href="#!">Support</a> }
          ]
        }}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
