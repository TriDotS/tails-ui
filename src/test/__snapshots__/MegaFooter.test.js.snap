// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mega footer snapshot 1`] = `
<div>
  <section
    className="bg-blue py-8 w-full"
  >
    <div
      className="container mx-auto px-8"
    >
      <div
        className="table w-full"
      >
        <div
          className="block sm:table-cell"
        >
          <p
            className="uppercase text-white font-bold text-md sm:mb-6"
          >
            Links
          </p>
          <ul
            className="list-reset text-xs mb-6"
          >
            <li
              className="mt-2 inline-block mr-2 sm:block sm:mr-0"
            >
              <a
                className="text-white hover:text-grey-light"
                href="#!"
              >
                FAQ
              </a>
            </li>
            <li
              className="mt-2 inline-block mr-2 sm:block sm:mr-0"
            >
              <a
                className="text-white hover:text-grey-light"
                href="#!"
              >
                About
              </a>
            </li>
            <li
              className="mt-2 inline-block mr-2 sm:block sm:mr-0"
            >
              <a
                className="text-white hover:text-grey-light"
                href="#!"
              >
                Support
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>
  <div
    className="bg-blue-dark text-xs text-center py-2 text-white"
  >
    Copyright 2017 Tyler Knipfer
  </div>
</div>
`;
